### PR TITLE
Internal Java solution for fine dining

### PR DESCRIPTION
**Pull request by nathangong**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [x] I have tested my code
- [x] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [x] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

File: solutions/usaco-861.mdx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class FineDining {
+	public static void main(String[] args) throws IOException {
+		Scanner sc = new Scanner(new File("dining.in"));
+		PrintWriter out = new PrintWriter("dining.out");
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-861.mdx:25]: Kattio?

**Review by Aadit-Ambadkar — APPROVED** [review]
Maybe Kattio but other than that great! Add an explanation maybe?

File: solutions/usaco-861.mdx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+		Edge(int other, int time) {
+			this.other = other;
+			this.time = time;
+		}
+	}
+
+	static class State implements Comparable<State> {
+		int pos, time;
+		int eaten;  // 1 if true, 0 if false
```
Comment (SansPapyrus683) [inline solutions/usaco-861.mdx:102]: why not just use a bool?

File: solutions/usaco-861.mdx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+			dist[i][0] = -1;
+			dist[i][1] = -1;
+		}
+
+		// run djiskstra's to find the shortest distances
+		PriorityQueue<State> pq = new PriorityQueue<>();
+		pq.add(new State(n - 1, 0, 0));
+		while (!pq.isEmpty()) {
+			State state = pq.remove();
+			if (dist[state.pos][state.eaten] != -1) continue;
```
Comment (SansPapyrus683) [inline solutions/usaco-861.mdx:64]: can you add braces for these one line control flows

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-861.mdx (hunk 2 above)
Comment (nathangong) [reply solutions/usaco-861.mdx:102]: easier to work with it as an int with compareTo and with the dp array

File: solutions/usaco-861.mdx (hunk 1 above)
Comment (nathangong) [reply solutions/usaco-861.mdx:25]: Scanner works fine in this problem, so there is no need for it

**Comment by nathangong** [comment]
> Maybe Kattio but other than that great! Add an explanation maybe?
 
 I think the official editorial does a good job for explaining the problem, but lmk if you disagree

File: solutions/usaco-861.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/usaco-861.mdx:104]: ok valid point

**Comment by Aadit-Ambadkar** [comment]
> > Maybe Kattio but other than that great! Add an explanation maybe?
> 
> I think the official editorial does a good job for explaining the problem, but lmk if you disagree

I think ur right! :D

**Comment by nathangong** [comment]
> use tabs for indentation
I'm pretty sure I did, is there anywhere that has spaces?

**Review by dongliu0426 — APPROVED** [review]

**Pull request merged**